hudi-client/src/main/java/org/apache/hudi/table/action/commit/CommitActionExecutor.java

```diff
@@ -89,11 +87,12 @@ public CommitActionExecutor(JavaSparkContext jsc,
       throw new HoodieUpsertException(
           "Error in finding the old file path at commit " + instantTime + " for fileId: " + fileId);
     } else {
-      AvroReadSupport.setAvroReadSchema(table.getHadoopConf(), upsertHandle.getWriterSchema());
       BoundedInMemoryExecutor<GenericRecord, GenericRecord, Void> wrapper = null;
-      try (ParquetReader<IndexedRecord> reader =
-          AvroParquetReader.<IndexedRecord>builder(upsertHandle.getOldFilePath()).withConf(table.getHadoopConf()).build()) {
-        wrapper = new SparkBoundedInMemoryExecutor(config, new ParquetReaderIterator(reader),
+      try {
+        HoodieStorageReader<IndexedRecord> storageReader =
+            HoodieStorageReaderFactory.getStorageReader(table.getHadoopConf(), upsertHandle.getOldFilePath());
+        wrapper =
+            new SparkBoundedInMemoryExecutor(config, storageReader.getRecordIterator(upsertHandle.getWriterSchema()),
```

Comment:
Assume this actually pushed the projection predicates down??

Reply:
This is just creating a ParquetReader and getting an iterator to read all the records. The entire record will need to be read here as we are merging.

I didn't understand how predicates are applicable here. Are they not in the InputFormat?

Reply:
No we specify a projection schema even without input format being involved in other places. E.g: in fetching keys for indexing

Reply:
@vinothchandar :  Guess its taken care of. Here is the getRecordIterator for Parquet.
```
public Iterator<R> getRecordIterator(Schema schema) throws IOException {
    AvroReadSupport.setAvroReadSchema(conf, schema);
    ParquetReader<IndexedRecord> reader = AvroParquetReader.<IndexedRecord>builder(path).withConf(conf).build();
    return new ParquetReaderIterator(reader);
  }
```
We set the scheme before instantiating the reader. 
Let me know is my understanding is wrong.